ftw/file/viewlets/content_history.pt

```diff
@@ -2,41 +2,53 @@
      tal:define="history view/fullHistory;"
      tal:condition="python:history and view.show_viewlet()"
      tal:omit-tag="">
-  <h2 i18n:translate="" class="contentHistoryTitle">Journal</h2>
-  <table class="vertical listing contentHistory">
-    <thead>
-      <tr>
-        <th i18n:translate="">Action</th>
-        <th i18n:translate="">Date</th>
-        <th i18n:translate="">Author</th>
-        <th i18n:translate="">Comment</th>
-        <th>&nbsp;</th>
-      </tr>
-    </thead>
-    <tbody>
-      <tr tal:repeat="item history">
-        <tal:def define="rhComments item/comments|nothing;
-                         actor item/actor;
-                         actor_name python:actor and actor.get('fullname','username') or item['actorid'];
-                         actor_home item/actor_home;
-                         action item/transition_title;
-                         isVersion python:item['type']=='versioning'">
-          <td>
-            <span class="historyAction" tal:content="action" i18n:translate=""
-                  tal:attributes="class string:historyAction state-${action}">action</span>
-          </td>
-          <td tal:content="python:view.toLocalizedTime(item['time'], long_format=True)" />
-          <td tal:content="actor_name" />
-          <td tal:content="item/comments" />
-          <td>
-            <a tal:condition="isVersion"
-               tal:attributes="href string:${context/absolute_url}/file_download_version?version_id=${item/version_id}"
-               i18n:translate="">
-              download
-            </a>
-          </td>
-        </tal:def>
-      </tr>
-    </tbody>
-  </table>
+  <div tal:define="Batch python:modules['Products.CMFPlone'].Batch;
```

Comment:
I have to add another nesting layer for batching. `define` is executed before `condition` and the batch code fails if `None` is given instead of a list (which is the case if there are no entrys).